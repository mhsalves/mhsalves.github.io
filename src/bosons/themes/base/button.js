import { css } from 'styled-components';

const normal = css`
  padding: 0.625rem 1.5rem;
  ${({ theme: { typography } }) => typography.normal}
`;

const small = css`
  padding: 0.375rem 1rem;
  ${({ theme: { typography } }) => typography.small}
`;

const common = css`
  border-radius: 2rem;
  color: ${({ theme: { colors } }) => colors.white.normal};
  text-transform: uppercase;
  cursor: pointer;

  ${({ isSmall }) => css`
    ${isSmall && small}
    ${!isSmall && normal}
  `}
`;

const primary = css`
  ${common}
  background-color: ${({ theme: { colors } }) => colors.primary.normal};

  &:hover {
    background-color: ${({ theme: { colors } }) => colors.primary.dark};
  }
`;

const secondary = css`
  ${common}
  background-color: ${({ theme: { colors } }) => colors.secondary.normal};

  &:hover {
    background-color: ${({ theme: { colors } }) => colors.secondary.dark};
  }
`;

export default {
  primary,
  secondary,
};
