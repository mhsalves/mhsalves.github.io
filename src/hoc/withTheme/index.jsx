import React from 'react';
import { ThemeProvider } from 'styled-components';
import themes from '../../bosons/themes';

const Wrapped = (Component) => (props) => (
  <ThemeProvider theme={themes.base}>
    <Component {...props} />
  </ThemeProvider>
);

export default Wrapped;
