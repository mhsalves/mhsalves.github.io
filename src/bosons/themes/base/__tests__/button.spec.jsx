import React from 'react';
import styled from 'styled-components';
import renderer from 'react-test-renderer';

import button from '../button';
import withTheme from '../../../../hoc/withTheme';

describe('testing: theme-base breakpoints', () => {
  test('should match primary button layout correctly', () => {
    const Button = styled.button`${button.primary}`;
    const ButtonWrapped = withTheme(Button);
    const buttonSnapshot = renderer.create(<ButtonWrapped />).toJSON();
    expect(buttonSnapshot).toMatchSnapshot();
  });
  test('should match secondary button layout correctly', () => {
    const Button = styled.button`${button.secondary}`;
    const ButtonWrapped = withTheme(Button);
    const buttonSnapshot = renderer.create(<ButtonWrapped />).toJSON();
    expect(buttonSnapshot).toMatchSnapshot();
  });
  test('should match small button size correctly', () => {
    const Button = styled.button`${button.primary}`;
    const ButtonWrapped = withTheme(Button);
    const buttonSnapshot = renderer.create(<ButtonWrapped isSmall />).toJSON();
    expect(buttonSnapshot).toMatchSnapshot();
  });
});
