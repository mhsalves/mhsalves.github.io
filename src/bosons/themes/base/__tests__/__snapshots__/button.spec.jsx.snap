// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing: theme-base breakpoints should match primary button layout correctly 1`] = `
.c0 {
  border-radius: 2rem;
  color: #FFFFFF;
  text-transform: uppercase;
  cursor: pointer;
  padding: 0.625rem 1.5rem;
  font-family: Nunito Sans,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  background-color: #3399FF;
}

.c0:hover {
  background-color: #2866A8;
}

@media (min-width:768px) {
  .c0 {
    font-size: 1rem;
    line-height: 1.5rem;
  }
}

<button
  className="c0"
/>
`;

exports[`testing: theme-base breakpoints should match secondary button layout correctly 1`] = `
.c0 {
  border-radius: 2rem;
  color: #FFFFFF;
  text-transform: uppercase;
  cursor: pointer;
  padding: 0.625rem 1.5rem;
  font-family: Nunito Sans,sans-serif;
  font-size: 1rem;
  line-height: 1.5rem;
  background-color: #6C757D;
}

.c0:hover {
  background-color: #5A6268;
}

@media (min-width:768px) {
  .c0 {
    font-size: 1rem;
    line-height: 1.5rem;
  }
}

<button
  className="c0"
/>
`;

exports[`testing: theme-base breakpoints should match small button size correctly 1`] = `
.c0 {
  border-radius: 2rem;
  color: #FFFFFF;
  text-transform: uppercase;
  cursor: pointer;
  padding: 0.375rem 1rem;
  font-family: Nunito Sans,sans-serif;
  font-size: .8125rem;
  line-height: 1rem;
  background-color: #3399FF;
}

.c0:hover {
  background-color: #2866A8;
}

@media (min-width:768px) {
  .c0 {
    font-size: .8125rem;
    line-height: 1rem;
  }
}

<button
  className="c0"
/>
`;
